src/helpers/PdfDownload/index.js

```diff
@@ -0,0 +1,38 @@
+export class PdfDownload {
+  constructor(event, form) {
+    this.submitEvt = event;
+    this.formInstance = form;
+  }
+
+  /**
+   * isPdfDownloadEnabled function check needPdf field is marked as 'yes' in form.io
+   * and then set PDF url as a session storage
+   * @return {undefined}
+   * */
+  isPdfDownloadEnabled() {
+    if (this.submitEvt.data && this.submitEvt.data.needPdf === "yes") {
+      const pdfUrl = this.submitEvt.metadata.pdfUrl.DownloadUrl;
+      window.sessionStorage.setItem("pdfUrl", pdfUrl);
+      this.formInstance.root.nextPage().then(() => {
```

Comment:
the logic here means the pdf download only controlled by the form.io form settings instead of matrix template attribute?

seems very different from what you've done previously

Reply:
Yes, it is controlled by forms.io so that we can use it in any project. If we just have options in Matrix then we can't use that in non matrix projects.